### PR TITLE
add missing check for static peers to allow them to exceed the connection limit

### PR DESCRIPTION
**Pull request by pinges**
## PR description
In one place there was a check missing to allow static peers to exceed the connection limit.
Another improvement is that streamBestPeers() does only return fully validated peers.

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java [hunk 1]
```diff
@@ -197,8 +195,11 @@ private boolean registerDisconnect(
         disconnectCallbacks.forEach(callback -> callback.onDisconnect(peer));
         peer.handleDisconnect();
         abortPendingRequestsAssignedToDisconnectedPeers();
-        LOG.debug("Disconnected EthPeer {}", peer.getShortNodeId());
-        LOG.trace("Disconnected EthPeer {}", peer);
+        if (peer.getReputation().getScore() > 102) {
```
Comment (Gabriel-Trintinalia) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:198]: Can you have this in a field? maybe something like: `USEFULL_PEER_SCORE_THRESHOLD`

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java [hunk 2]
```diff
@@ -220,12 +221,8 @@ private void abortPendingRequestsAssignedToDisconnectedPeers() {
...
-      return incompleteConnections.get(
-          connection, () -> completeConnections.get(connection.getPeer().getId()));
-    } catch (final ExecutionException e) {
-      throw new RuntimeException(e);
-    }
+    final EthPeer ethPeer = incompleteConnections.getIfPresent(connection);
+    return ethPeer != null ? ethPeer : completeConnections.get(connection.getPeer().getId());
   }
 
   public EthPeer peer(final Bytes peerId) {
```
Comment (Gabriel-Trintinalia) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:228]: Is this method still being used?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java [hunk 3]
```diff
@@ -321,7 +318,9 @@ public Stream<EthPeer> streamAvailablePeers() {
   }
 
   public Stream<EthPeer> streamBestPeers() {
-    return streamAvailablePeers().sorted(getBestChainComparator().reversed());
+    return streamAvailablePeers()
+        .filter(EthPeer::isFullyValidated)
```
Comment (Gabriel-Trintinalia) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:322]: can we add this to the description of the PR?

**Review by Gabriel-Trintinalia — CHANGES_REQUESTED** [review]

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/PeerPrivileges.java [hunk 4]
```diff
@@ -23,5 +25,5 @@ public interface PeerPrivileges {
    * @param peer The peer to be checked.
```
Comment (Gabriel-Trintinalia) [inline ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/PeerPrivileges.java:25]:
```suggestion
   * @param peerId The peer to be checked.
```

**Review by Gabriel-Trintinalia — CHANGES_REQUESTED** [review]

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java (hunk 1 above)
Comment (pinges) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:198]: Nice!

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java (hunk 3 above)
Comment (pinges) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:322]: Done

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java (hunk 2 above)
Comment (pinges) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:228]: It is not! Removed!

File: ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/PeerPrivileges.java (hunk 4 above)
Comment (pinges) [reply ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/PeerPrivileges.java:25]: Done

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java [hunk 5]
```diff
@@ -197,8 +196,11 @@ private boolean registerDisconnect(
         disconnectCallbacks.forEach(callback -> callback.onDisconnect(peer));
         peer.handleDisconnect();
         abortPendingRequestsAssignedToDisconnectedPeers();
-        LOG.debug("Disconnected EthPeer {}", peer.getShortNodeId());
-        LOG.trace("Disconnected EthPeer {}", peer);
+        if (peer.getReputation().getScore() > USEFULL_PEER_SCORE_THRESHOLD) {
+          LOG.debug("Disonnected USEFULL peer {}", peer);
+        } else if (!LOG.isTraceEnabled()) {
```
Comment (Gabriel-Trintinalia) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:201]: We are checking for isTraceEnabled but logging at the debug level, should we make it a trace?

**Review by Gabriel-Trintinalia — APPROVED** [review]

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java (hunk 5 above)
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthPeers.java:201]: I missed the "!", is that on purpose? Do we log only if Trace is not enabled?

**Pull request merged**